svir/dialogs/ipt_dialog.py

```diff
@@ -0,0 +1,291 @@
+from qgis.PyQt.QtNetwork import QNetworkRequest, QHttpMultiPart, QHttpPart
+from svir.dialogs.standalone_app_dialog import StandaloneAppDialog, GemApi
+from svir.utilities.shared import DEBUG, REQUEST_ATTRS
+
+
+class IptDialog(StandaloneAppDialog):
+    """
+    Dialog that embeds the OpenQuake Input Modelling Toolkit
+    standalone application
+    """
+
+    def __init__(self, ipt_dir, irmt, parent=None):
+        self.irmt = irmt
+        self.ipt_dir = ipt_dir
+        app_name = 'ipt'
+        app_descr = 'OpenQuake Input Preparation Toolkit'
+        gem_header_name = "Gem--Qgis-Oq-Irmt--Ipt"
+        gem_header_value = "0.1.0"
+        super(IptDialog, self).__init__(
+            app_name, app_descr, gem_header_name, gem_header_value, parent)
+        self.gem_api = IptPythonApi(self.message_bar, self)
+        self.build_gui()
+
+    def build_gui(self):
+        super(IptDialog, self).build_gui()
+        if DEBUG:
+            self.set_example_btn = QPushButton("Set example")
+            self.set_example_btn.clicked.connect(
+                self.on_set_example_btn_clicked)
+            ipt_example = '%s/%s?tab_id=1&subtab_id=0&example_id=99' % (
+                self.host, self.app_name)
+            self.example_url = QLineEdit(ipt_example)
+            self.hlayout = QHBoxLayout()
+            self.hlayout.addWidget(self.example_url)
+            self.hlayout.addWidget(self.set_example_btn)
+            self.vlayout.addLayout(self.hlayout)
+
+    def on_set_example_btn_clicked(self):
+        qurl = QUrl(self.example_url.text())
+        self.web_view.load(qurl)
+
+
+class IptPythonApi(GemApi):
+    """
+    API methods that are specific for the IPT application
+    (other shared methods are defined in the CommonApi)
+    """
+
+    @pyqtSlot(result='QVariantMap')
+    def select_file(self):
+        """
+        Open a file browser to select a single file in the ipt_dir,
+        and return the name of the selected files
+        """
+        try:
+            ipt_dir = self.parent().ipt_dir
+            file_name = QFileDialog.getOpenFileName(
+                self.parent().parent(), 'Select file', ipt_dir)
+            basename = os.path.basename(file_name)
+        except Exception as exc:
+            return {'ret': 1, 'reason': str(exc)}
+        else:
+            return {'ret': basename, 'reason': 'ok'}
+
+    @pyqtSlot(result='QVariantMap')
+    def select_files(self):
+        """
+        Open a file browser to select multiple files in the ipt_dir,
+        and return the list of names of selected files
+        """
+        try:
+            ipt_dir = self.parent().ipt_dir
+            file_names = QFileDialog.getOpenFileNames(
+                self.parent().parent(), 'Select files', ipt_dir)
+            ls = [os.path.basename(file_name) for file_name in file_names]
+        except Exception as exc:
+            return {'ret': 1, 'reason': str(exc)}
+        else:
+            return {'ret': ls, 'reason': 'ok'}
```

Comment:
Seems to me not consistent the return semantic, ret change type if success or not ... not so comfortable.
could be always an integer and 2 different other attribute in case of success and error ('content' and 'reason')

Reply:
I agree